HOWTO_RELEASE.md

```diff
@@ -0,0 +1,169 @@
+
+## 3. Make a release branch off `master` and draft the release
+
+When all tests succeed and the docs are fine, start a release branch. All work until the creation of the git release tag is now done on the release branch.
+**Do not touch any code on this branch other than indicated below!**
+
+- Create a branch off current `master` with the name `release-X.Y.Z`, inserting the correct version number, of course.
+- Compile the release notes. They should contain all *user-visible* changes (internal stuff like modified tests is not required) and should be summarized in one or two sentences on top, perhaps mentioning the most notable changes. Check the [Release Notes](https://github.com/odlgroup/odl/blob/master/doc/source/release_notes.rst) file for details on sections, formatting etc.
+- Increment the version number. The current locations of version numbers are the [odl/__init__.py](https://github.com/odlgroup/odl/blob/master/odl/__init__.py) and [conda/meta.yaml](https://github.com/odlgroup/odl/blob/master/conda/meta.yaml). To make sure you don't miss any other location (or the information here is outdated), perform a search:
+  ```
+  $ cd doc && make clean && cd ..  # remove the local HTML doc first
+  $ grep -Ir "0\.5\.4" . | grep -E -v "\.git|release_notes\.rst|odl\.egg-info"
+  ```
+- Change the `git_rev` field in `conda/meta.yaml` to `'vX.Y.Z'`, using the upcoming version number. This is the git tag you will create when making the release on GitHub.
+- Commit the changes, using a message like `REL: bump version to X.Y.Z`.
+- Make a PR and merge it after review.
+
+## 4. Make a release on GitHub
+
+Now that the version is incremented,
+
+- make a new [Release](https://github.com/odlgroup/odl/releases) on GitHub.
+- Paste the short summary from the release notes file (converting from RST to Markdown) but don't insert the details.
+- Add a link to the current section in the release notes file.
+
+## 5. Create packages for PyPI and Conda
+
+It is important to **update the local git repository** before creating packages, to make sure that they are built from the correct git revision.
+
+- Pull from GitHub and switch to `master` (or to the new version tag, which should be equivalent):
+  ```
+  $ git checkout master
+  $ git pull
+  ```
+
+- Making the packages for PyPI is straightforward. However, **make sure you delete old `build` directories** since they can pollute new builds:
+  ```
+  $ rm build/ -rf
+  $ python setup.py sdist
+  $ python setup.py bdist_wheel
+  ```
+  The packages are by default stored in a `dist` folder.
+
+- To build the conda packages, you should *not* work in a specific environment but rather exit to the root environment. There, install the `conda-build` tool for building packages:
+  ```
+  $ source deactivate
+  $ conda install conda-build
+  ```
+- Invoke the following command to build a package for your platform and all supported Pyhton versions:
```

Comment:
It also seems there is a "conda build --py all" option, that does what you want here